sympy/functions/combinatorial/factorials.py

```diff
@@ -231,7 +231,7 @@ def _eval_Mod(self, q):
                     else:
                         fc = self._facmod(n, aq)
 
-                    return Integer(fc % q)
+                    return S(fc % q)
```

Comment:
This change is not needed, keep the `Integer`, same as below.

Reply:
I think this is needed, otherwise, [this](https://github.com/sympy/sympy/blob/f3c949f043adc907179b5dd12e736ecf9a75a9c6/sympy/functions/combinatorial/tests/test_comb_factorials.py#L226) test fails.

Reply:
Sorry, the test just above that